\subsubsection{SEQUENCE MANIPULATION}
The Sequence Manipulation is a collection of web-based programs for analyzing and formatting DNA and protein sequences.
The output of each program is a set of HTML commands, which is rendered by your web browser as a standard web page. You can print and save the results, and you can edit them using an HTML editor or a text editor.
It is also used to remove non coding character and to get reverse and complement strand and also to calculate gc content.

\subsubsection{DATA MINING}
Data Mining tools are software programs that help in framing and executing data mining techniques to create data models and test them as well. It is usually a framework like R studio or Tableau with a suite of programs to help build and test a data model.
Data mining is the process of sorting through large data sets to identify patterns and relationships that can help solve business problems through data analysis. Data mining techniques and tools enable enterprises to predict future trends and make more-informed business decisions.
Data mining is used to explore increasingly large databases and to improve market segmentation. By analysing the relationships between parameters such as customer age, gender, tastes, etc., it is possible to guess their behaviour in order to direct personalised loyalty campaigns.

\subsubsection{MELTING TEMPERATURE CALCULATOR}

It is used to calculate melting temperature of an oligonucleotide.
It is  designed to give researchers an easy tool for finding and comparing melting temperatures using more accurate calculations.
For sequences less than 14 nucleotides the formula is: Tm= (wA+xT) * 2 + (yG+zC) * 4. where w,x,y,z are the number of the bases A,T,G,C in the sequence, respectively.
For sequences longer than 13 nucleotides, the equation used is. Tm= 64.9 +41*(yG+zC-16.4)/(wA+xT+yG+zC)


\subsubsection{PCR AMPLIFICATION}
The tool is used to stimulate PCR amplification and allow one mismatch between primer and template.
The entire cycling process of PCR is automated and can be completed in just a few hours. It is directed by a machine called a thermocycler, which is programmed to alter the temperature of the reaction every few minutes to allow DNA denaturing and synthesis
PCR is used in molecular biology to make many copies of (amplify) small sections of DNA or a gene. Using PCR it is possible to generate thousands to millions of copies of a particular section of DNA from a very small amount of DNA. PCR is a common tool used in medical and biological research labs
Thermocycler is used to complete a PCR amplification. The thermocycler can be set to cycle through the denaturation, annealing, and extension stages in a determined amount of cycles based on the reaction volume.
.

\subsubsection{MICROSATELLITE REPEATS FINDER}

The tool is used to find microsatellite in DNA sequence.
The most common microsatellite motifs in the human genome are A/T rich and more than a third of microsatellites in our data set (36.4%) are composed of the motifs A/T or AC/GT (Table 1). These two motifs are also the most common motifs within 5 kb of the TSS 
A microsatellite is a tract of repetitive DNA in which certain DNA motifs (ranging in length from one to six or more base pairs) are repeated, typically 5–50 times.
They are widely used for DNA profiling in cancer diagnosis, in kinship analysis (especially paternity testing) and in forensic identification. They are also used in genetic linkage analysis to locate a gene or a mutation responsible for a given trait or disease. Microsatellites are also used in population genetics to measure levels of relatedness between subspecies, groups and individuals

\subsubsection{PALINDROMIC SEQUENCE FINDER}

This tool allows selection of minimum and maximum size of palindromic sequence.
A DNA sequence whose 5'-to-3' sequence is identical on each DNA strand. The sequence is the same when one strand is read left to right and the other strand is read right to left. Recognition sites of many restriction enzymes are palindromic.
For a nucleotide sequence to be considered as a palindrome, its complementary strand must read the same in the opposite direction  For example, the sequence 5'-CGATCG-3' is considered a palindrome since its reverse complement 3'-GCTAGC-5' reads the same.
A palindromic sequence of nucleotides (which are labeled A, T, C, or G) occurs when complementary strands of DNA read the same in both directions, either from the 5-prime end or the 3-prime end.

\subsubsection{SEQUENCE ALIGNMENT}

The tool is used to allign two DNA ,RNA or protein sequence In this method SMITH-WATERMAN alignment is used.
In bioinformatics, a sequence alignment is a way of arranging the sequences of DNA, RNA, or protein to identify regions of similarity that may be a consequence of functional, structural, or evolutionary relationships between the sequences.
Very short or very similar sequences can be aligned by hand. However, most interesting problems require the alignment of lengthy, highly variable or extremely numerous sequences that cannot be aligned solely by human effort. Instead, human knowledge is applied in constructing algorithms to produce high-quality sequence alignments, and occasionally in adjusting the final results to reflect patterns that are difficult to represent algorithmically (especially in the case of nucleotide sequences). Computational approaches to sequence alignment generally fall into two categories: global alignments and local alignments.

\subsubsection{RESTRICTION DIGEST OF DNA}

The tool is used to restrict digestion of DNA sequence with endonucleases.
Restriction Digestion is the process of cutting DNA molecules into smaller pieces with special enzymes called Restriction Endonucleases (sometimes just called Restriction Enzymes or RE's).
Restriction enzyme digestion is commonly used in molecular cloning techniques, such as PCR or restriction cloning. It is also used to quickly check the identity of a plasmid by diagnostic digest.
Restriction enzymes, also called restriction endonucleases, recognize a specific sequence of nucleotides in double stranded DNA and cut the DNA at a specific location. They are indispensable to the isolation of genes and the construction of cloned DNA molecules.

\subsubsection{TRANSLATE DNA TO PROTEIN}

The tool is used to translate DNA sequence to protein by using genetic code including customized ones.
Basically, a gene is used to build a protein in a two-step process:
Step 1: transcription! Here, the DNA sequence of a gene is "rewritten" in the form of RNA. ...
Step 2: translation! In this stage, the mRNA is "decoded" to build a protein (or a chunk/subunit of a protein) that contains a specific series of amino acids.
The flow of information from DNA to RNA to proteins is one of the fundamental principles of molecular biology. It is so important that it is sometimes called the “central dogma.” Through the processes of transcription and translation, information from genes is used to make proteins


\subsubsection{TRANSLATE PROTEIN TO DNA}

The tool is used to obtain reverse translation of protein sequence.
Reverse Translate accepts a protein sequence as input and uses a codon usage table to generate a DNA sequence representing the most likely non-degenerate coding sequence. A consensus sequence derived from all the possible codons for each amino acid is also returned. Use Reverse Translate when designing PCR primers to anneal to an unsequenced coding sequence from a related species.

\subsubsection{RANDOM SEQUENCE}
The tool is used to generate random DNA or protein sequence.
The concept of a random sequence is essential in probability theory and statistics. The concept generally relies on the notion of a sequence of random variables and many statistical discussions begin with the words "let X1,...,Xn be independent random variables...". Yet as D. H. Lehmer stated in 1951: "A random sequence is a vague notion... in which each term is unpredictable to the uninitiated and whose digits pass a certain number of tests traditional with statisticians
this tool allows user to generate randomized sequences of integers. The randomness comes from atmospheric noise, which for many purposes is better than the pseudo-random number algorithms typically used in computer programs.

\subsubsection{OLIGONUCLEOTIDE FREQUENCY}

The tool is used to count number of occurences of each nucleoide with length between 1 and 8 within sequence.
Given a DNA or RNA sequence (or a set of DNA or RNA sequences), the oligonucleotideFrequency function computes the frequency of all possible oligonucleotides of a given length (called the "width" in this particular context) in a sliding window that is shifted step nucleotides at a time.
The dinucleotideFrequency and trinucleotideFrequency functions are convenient wrappers for calling oligonucleotideFrequency with width=2 and width=3, respectively.
The nucleotideFrequencyAt function computes the frequency of the short sequences formed by extracting the nucleotides found at some fixed positions from each sequence of a set of DNA or RNA sequences.

